src/ucai/core/utils/callable_utils.py

```diff
@@ -0,0 +1,361 @@
+    """
+    if hasattr(param_hint, '__origin__'):
+        origin = param_hint.__origin__
+        if origin == list:
+            return (
+                f"Error in parameter '{param_name}': List type requires a specific element type. "
+                f"Please define the internal type for the list, e.g., List[int]. Original error: {exception}"
+            )
+        elif origin == tuple:
+            return (
+                f"Error in parameter '{param_name}': Tuple type requires specific element types. "
+                f"Please define the types for the tuple elements, e.g., Tuple[int, str]. Original error: {exception}"
+            )
+        elif origin == dict:
+            return (
+                f"Error in parameter '{param_name}': Dict type requires both key and value types. "
+                f"Please define the internal types for the dict, e.g., Dict[str, int]. Original error: {exception}"
+            )
+    return (
+        f"Error in parameter '{param_name}': type {param_hint} is not supported. "
+        f"Original error: {exception}"
+    )
+
+def format_default_value(default: Any) -> str:
+    """Formats the default value for SQL."""
+    if default is None:
+        return 'NULL'
+    elif isinstance(default, str):
+        return f"'{default}'"
+    else:
+        return str(default)
+
+def unwrap_function(func: Callable[..., Any]) -> Callable[..., Any]:
+    """Unwraps staticmethod or classmethod to get the actual function."""
+    if isinstance(func, (staticmethod, classmethod)):
+        func = func.__func__
+    return func
+
+def process_parameter(
+    param_name: str,
+    param: inspect.Parameter,
+    type_hints: dict[str, Any],
+    docstring_info: DocstringInfo
+) -> str:
+    """Processes a single parameter and returns its SQL definition."""
+    if param_name in FORBIDDEN_PARAMS:
+        raise ValueError(f"Parameter '{param_name}' is not allowed in the function signature.")
+
+    if param.kind in (inspect.Parameter.VAR_POSITIONAL, inspect.Parameter.VAR_KEYWORD):
+        kind = 'var-positional (*args)' if param.kind == inspect.Parameter.VAR_POSITIONAL else 'var-keyword (**kwargs)'
+        raise ValueError(f"Parameter '{param_name}' is a {kind} parameter, which is not supported in SQL functions.")
+
+    if param_name not in type_hints:
+        raise ValueError(f"Missing type hint for parameter: {param_name}.")
+
+    param_hint = type_hints[param_name]
+
+    try:
+        sql_type = validate_type_hint(param_hint)
+    except ValueError as e:
+        error_message = generate_type_hint_error_message(param_name, param_hint, e)
+        raise ValueError(error_message) from e
+
+    param_comment = docstring_info.params.get(param_name, f"Parameter {param_name}").replace("'", '"')
+
+    if param.default is inspect.Parameter.empty:
+        return f"{param_name} {sql_type} COMMENT '{param_comment}'"
+    else:
+        default_value = format_default_value(param.default)
```

Comment:
Should we validate the param.default value is of correct type? 
For example, below code actually works:
```
def test(a: List[str]=1):
    return a
test()
```